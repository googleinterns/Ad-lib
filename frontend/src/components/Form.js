import React from 'react';
import axios from 'axios';
import 'date-fns';
import DateFnsUtils from '@date-io/date-fns';
import {MuiPickersUtilsProvider,
  KeyboardTimePicker} from '@material-ui/pickers';
import {makeStyles} from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import MatchPreference from './MatchPreference';
import Checkbox from './PreferencesCheckbox';
import RoleDropdown from './RoleDropdown';
import ProductAreaDropdown from './ProductAreaDropdown';
import DurationDropdown from './DurationDropdown';

/** Establishes style to use on rendering form component */
const useStyles = makeStyles((theme) => ({
  flexStartDiv: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    margin: theme.spacing(2),
  },
  flexEndDiv: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
  section: {
    margin: theme.spacing(2),
    width: 650,
  },
  heading: {
    marginLeft: theme.spacing(2),
  },
  padding: {
    marginTop: theme.spacing(4),
    marginLeft: theme.spacing(2),
  },
}));

/**
 * Validates from inputs on submission and alerts user on error
 * @return {Boolean}  true or false based on validity of inputs
 * @param {String} role
 * @param {String} productArea
 * @param {Number} duration
 * @param {Date} timeAvailableUntil
 * @param {Number} currentTimeMilliseconds
 */
export function validateFormInputs(role, productArea, duration,
    timeAvailableUntil, currentTimeMilliseconds) {
  if (!(timeAvailableUntil instanceof Date) ||
    isNaN(timeAvailableUntil.getTime())) {
    alert('Please select a valid date.');
    return false;
  }
  const timeAvailableUntilMilliseconds = timeAvailableUntil.getTime();
  const durationMilliseconds = duration * 60000;
  if (currentTimeMilliseconds + durationMilliseconds >=
    timeAvailableUntilMilliseconds) {
    // Check if a meeting is possible with the provided inputs
    alert('Please select an larger time availability window');
    return false;
  } else if (productArea === '' || role === '') {
    alert('Please select options for all form fields to submit!');
    return false;
  } else {
    return true;
  }
}

/**
 * Create form component with time and match preference inputs
 * @return {Form} Form component
 */
export default function Form() {
  const classes = useStyles();

  // Declare state variables for each input field and set default states
  const [timeAvailableUntil, setTimeAvailableUntil] =
    React.useState(new Date());
  const [duration, setDuration] = React.useState(15);
  const [productArea, setProductArea] = React.useState('');
  const [role, setRole] = React.useState('');
  const [savePreference, setSavePreference] = React.useState(true);
  const [matchPreference, setMatchPreference] = React.useState('none');

  /** Gather user inputs from form and send POST request to backend
    * @param {Event} event
   */
  function handleFormSubmission(event) {
    const currentTimeInMilliseconds = new Date().getTime();
    if (validateFormInputs(role, productArea, duration,
        timeAvailableUntil, currentTimeInMilliseconds)) {
      // Override browser's default behvaior to execute POST request
      event.preventDefault();

      // Gather all form inputs into one object
      const formDetails = {
        timeAvailableUntil: timeAvailableUntil.getTime(),
        duration: duration,
        role: role,
        productArea: productArea,
        matchPreference: matchPreference,
        savePreference: savePreference,
      };

      // Send form details to AddParticipantServlet and alert user on success
      axios.post('/api/v1/add-participant', {formDetails})
          .then((response) => {
            if (response.data != null) {
              // TODO(#33): change alert to a redirection to loading page view
              alert('Successful');
            }
          });
    }
  }

  return (
    <div>
      <div className={classes.heading}>
        <h3>Choose your time preferences</h3>
      </div>
      <div className={classes.section}>
        <div className={classes.flexStartDiv}>
          <p>I am free until...</p>
          <div style={{width: 180}}>
            <MuiPickersUtilsProvider utils={DateFnsUtils}>
              <KeyboardTimePicker
                id="time-field"
                value={timeAvailableUntil}
                onChange={(value) => setTimeAvailableUntil(value)}
                KeyboardButtonProps={{'aria-label': 'time-field'}}
              />
            </MuiPickersUtilsProvider>
          </div>
        </div>
        <div className={classes.flexStartDiv}>
          <p>I want to talk for...</p>
          <DurationDropdown onChange={(value) => setDuration(value)} />
        </div>
      </div>
      <div className={classes.heading}>
        <h3>Choose your match preferences</h3>
      </div>
      <div className={classes.section}>
        <div className={classes.flexStartDiv}>
          <RoleDropdown onChange={(value) => setRole(value)} />
          <ProductAreaDropdown onChange={(value) => setProductArea(value)} />
        </div>
        <div className={classes.padding}>
          <MatchPreference onChange={(value) => setMatchPreference(value)} />
        </div>
      </div>
      <div className={classes.flexEndDiv}>
        <Checkbox onChange={(value) => setSavePreference(value)} />
        <div className={classes.flexEndDiv}>
          <Button variant="contained" color="primary"
            onClick={handleFormSubmission}>
            Submit
          </Button>
        </div>
      </div>
    </div>
  );
}
