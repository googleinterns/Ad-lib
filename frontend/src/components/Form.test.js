import React from 'react';
import Form from './Form';
import renderer from 'react-test-renderer';
import axios from 'axios';
import mockAxios from 'axios';
import {validateFormInputs} from './Form';

beforeAll(() => {
  const DATE_TO_USE = new Date('2020');
  const mockedDate = Date;
  global.Date = jest.fn(() => DATE_TO_USE);
  global.Date.UTC = mockedDate.UTC;
  global.Date.parse = mockedDate.parse;
  global.Date.now = mockedDate.now;
});

jest.mock('axios');

describe('Form', () => {
  it('should be defined', () => {
    expect(Form).toBeDefined();
  });

  it('should render correctly in varying times and timezones', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('POST request using axios() to servlet with form details', () => {
    const mockedDate = new Date();
    const servletEndpoint = '/api/v1/add-participant';
    const mockFormDetails = {
      timeAvailableUntil: mockedDate.getTime(),
      duration: 15,
      role: 'Intern',
      productArea: 'Core',
      matchPreference: 'similar',
      savePreference: true,
    };

    axios.post(servletEndpoint, {mockFormDetails});

    expect(mockAxios.post).toHaveBeenCalledTimes(1);
    expect(mockAxios.post).toHaveBeenCalledWith(
        servletEndpoint, {mockFormDetails},
    );
  });
});

describe('Form Validation', () => {
  it('should return true', () => {
    window.alert = jest.fn();
    const role = 'Intern';
    const productArea = 'Platforms and Ecosystems';
    const timeAvailableUntilInMilliseconds = 1594846605591;
    const duration = 15;
    const currentTimeInMilliseconds = 1594801800000;
    expect(validateFormInputs(role, productArea, duration,
        timeAvailableUntilInMilliseconds, currentTimeInMilliseconds)).toBe(true);
  });

  it('should return false since Role is not specified', () => {
    window.alert = jest.fn();
    const role = '';
    const productArea = 'Platforms and Ecosystems';
    const timeAvailableUntilInMilliseconds = 1594846605591;
    const duration = 15;
    const currentTimeInMilliseconds = 1594801800000;
    expect(validateFormInputs(role, productArea, duration,
        timeAvailableUntilInMilliseconds, currentTimeInMilliseconds)).toBe(false);
  });

  it('should return false since PA is not specified', () => {
    window.alert = jest.fn();
    const role = 'Intern';
    const productArea = '';
    const timeAvailableUntilInMilliseconds = 1594846605591;
    const duration = 15;
    const currentTimeInMilliseconds = 1594801800000;
    expect(validateFormInputs(role, productArea, duration,
        timeAvailableUntilInMilliseconds, currentTimeInMilliseconds)).toBe(false);
  });

  it('should return false since time is incompatible', () => {
    window.alert = jest.fn();
    const role = 'Intern';
    const productArea = 'Core';
    const timeAvailableUntilInMilliseconds = 1594846605591;
    const duration = 15;
    const currentTimeInMilliseconds = 1594846605000;
    expect(validateFormInputs(role, productArea, duration,
        timeAvailableUntilInMilliseconds, currentTimeInMilliseconds)).toBe(false);
  });
});
