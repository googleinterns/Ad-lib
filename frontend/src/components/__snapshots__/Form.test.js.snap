// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly in varying times and timezones 1`] = `
<div>
  <div
    className="makeStyles-heading-4"
  >
    <h3>
      Choose your time preferences
    </h3>
  </div>
  <div
    className="makeStyles-section-3"
  >
    <div
      className="makeStyles-flexStartDiv-1"
    >
      <p>
        I am free until...
      </p>
      <div
        style={
          Object {
            "width": 180,
          }
        }
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              disabled={false}
              id="time-field"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="12:00 AM"
            />
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="time-field"
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
          
        </div>
      </div>
    </div>
    <div
      className="makeStyles-flexStartDiv-1"
    >
      <p>
        I want to talk for...
      </p>
      <div>
        <div
          className="MuiFormControl-root"
          style={
            Object {
              "width": 180,
            }
          }
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <select
              aria-invalid={false}
              aria-label="duration"
              className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
              disabled={false}
              id="duration-input"
              name="duration"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
            >
              <option
                value={15}
              >
                15 minutes
              </option>
              <option
                value={30}
              >
                30 minutes
              </option>
              <option
                value={45}
              >
                45 minutes
              </option>
              <option
                value={60}
              >
                60 minutes
              </option>
            </select>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiNativeSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-heading-4"
  >
    <h3>
      Choose your match preferences
    </h3>
  </div>
  <div
    className="makeStyles-section-3"
  >
    <div
      className="makeStyles-flexStartDiv-1"
    >
      <div>
        <div
          className="MuiFormControl-root"
          style={
            Object {
              "width": 180,
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            id="role"
          >
            Role
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-label="role"
              aria-labelledby="role-input"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="role-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              name="role"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div>
        <div
          className="MuiFormControl-root"
          style={
            Object {
              "width": 180,
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            id="productArea"
          >
            Product Area
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-label="productArea"
              aria-labelledby="productArea-input"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="productArea-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              name="productArea"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div>
        <div
          className="MuiFormControl-root"
          style={
            Object {
              "width": 180,
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            id="interests"
          >
            Interests
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-label="interests"
              aria-labelledby="interests"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="interests"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-padding-5"
    >
      <fieldset
        className="MuiFormControl-root"
      >
        <div
          aria-label="matchPreference"
          className="MuiFormGroup-root MuiFormGroup-row"
          defaultValue="none"
          role="radiogroup"
        >
          <label
            className="MuiFormControlLabel-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input-9"
                  disabled={false}
                  name="matchPreference"
                  onChange={[Function]}
                  type="radio"
                  value="similar"
                />
                <div
                  className="PrivateRadioButtonIcon-root-10"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-11"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
            </span>
            <span
              className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Similar Googler
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-checked-7 Mui-checked MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  checked={true}
                  className="PrivateSwitchBase-input-9"
                  disabled={false}
                  name="matchPreference"
                  onChange={[Function]}
                  type="radio"
                  value="none"
                />
                <div
                  className="PrivateRadioButtonIcon-root-10 PrivateRadioButtonIcon-checked-12"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-11"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
            </span>
            <span
              className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              No Preference
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input-9"
                  disabled={false}
                  name="matchPreference"
                  onChange={[Function]}
                  type="radio"
                  value="different"
                />
                <div
                  className="PrivateRadioButtonIcon-root-10"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-11"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
            </span>
            <span
              className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Different Googler
            </span>
          </label>
        </div>
      </fieldset>
    </div>
  </div>
  <div
    className="makeStyles-flexEndDiv-2"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-7 Mui-checked MuiIconButton-colorPrimary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-9"
            data-indeterminate={false}
            defaultChecked={true}
            name="checkbox"
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Save my preferences for later
      </span>
    </label>
    <div
      className="makeStyles-flexEndDiv-2"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </div>
  </div>
</div>
`;
